apps/builder/app/builder/features/props-panel/props-panel.tsx

```diff
@@ -101,34 +135,31 @@ const Property = ({
   onChange: (value: PropValue) => void;
   onDelete?: () => void;
   setCssProperty: SetCssProperty;
-}) => (
-  <Box css={{ mb: theme.spacing[9] }}>
-    {renderControl({
-      meta,
-      prop,
-      propName,
-      onDelete,
-      onChange: (propValue, asset) => {
-        onChange(propValue);
-
-        // @todo: better way to do this?
-        if (
-          component === "Image" &&
-          propName === "src" &&
-          asset &&
-          "width" in asset.meta &&
-          "height" in asset.meta
-        ) {
-          setCssProperty("aspectRatio")({
-            type: "unit",
-            unit: "number",
-            value: asset.meta.width / asset.meta.height,
-          });
-        }
-      },
-    })}
-  </Box>
-);
+}) =>
+  renderControl({
+    meta,
+    prop,
+    propName,
+    onDelete,
+    onChange: (propValue, asset) => {
+      onChange(propValue);
+
+      // @todo: better way to do this?
```

Comment:
looks like we could make a better API for type of thing, I bet there is a name for the pattern, it almost looks like a transformer/map but instead it sets the property

Reply:
not for right now, but we most likely will have more of these in the future

Reply:
Yeah, will need to think how to solve this later. Don't want to do this now. This is not new code added in this PR.